course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -1,13 +1,44 @@
+// regex taken from https://www.ietf.org/rfc/rfc3986.txt appendix B, which
```

Comment:
Short answer: This is only used to test a url generated by our testing, so as long as it works, I'm happy with it.  If this were user-facing code, I'd be paranoid enough to pull in a utility library that has some test coverage to back it up.

Long answer: Per https://www.rfc-editor.org/info/rfc3986, there are two "updated by" RFCs, one referencing ipv6 (which the regex will fail on), and one offering suggestions on restricting uri definitions based on the scheme (we only care about http/https).  There is an erratum correcting the regex syntax, including escaping, so I've switched to using that one.